File: LeetCodeProblemsCMake/Problems/LeetCode2176CountEqualAndDivisiblePairsInAnArray.cpp
//
// Created by saeed on 4/17/25.
//

#include "LeetCode2176CountEqualAndDivisiblePairsInAnArray.h"

int LeetCode2176CountEqualAndDivisiblePairsInAnArray::countPairs(std::vector<int> &nums, int k) {
    int length = static_cast<int>(nums.size());
    int result = 0;
    for (int i = 0; i < length-1; i++) {
        for (int j = i+1; j < length; j++) {
            if (nums[i] != nums[j])
                continue;
            if ((i*j)%k!=0)
                continue;
            result ++;
        }
    }
    return result;
}

File: LeetCodeProblemsGo/LeetCode2176CountEqualAndDivisiblePairsInAnArray/LeetCode2176CountEqualAndDivisiblePairsInAnArray.go
package LeetCode2176CountEqualAndDivisiblePairsInAnArray

func countPairs(nums []int, k int) int {
	length := len(nums)
	result := 0
	for i := 0; i < length-1; i++ {
		for j := i + 1; j < length; j++ {
			if nums[i] != nums[j] {
				continue
			}
			if (i*j)%k != 0 {
				continue
			}
			result++
		}
	}
	return result
}
